### PR TITLE
Verify callbacks with no docs included

### LINKED ISSUE #447: Callbacks with no docs doesn't show
Only callbacks with `@doc false` should be hidden. See `Phoenix.Socket` for an example.

### PR DESCRIPTION
**Pull request by bjfish**
The issue #447 does not appear to be broken anymore and could be closed.

A test is included in this PR to verify this is fixed in case you would like to include a test for this.

Fixes #447

### CODE REVIEW

File: test/fixtures/callbacks_no_docs.ex [hunk 1]
```diff
@@ -0,0 +1,7 @@
+defmodule CallbacksNoDocs do
+  use Behaviour
```
Comment (josevalim) [inline test/fixtures/callbacks_no_docs.ex]: Behaviour is deprecated. Can you please use `@callback` instead of `defcallback`?

**Comment by bjfish** [comment]
This fixed now to use `@callback`

**Comment by josevalim** [comment]
The build is failing: https://github.com/elixir-lang/ex_doc/pull/527 it seems to be an Elixir version issue then? Which version do you have?

**Comment by bjfish** [comment]
My Elixir version is `1.2.3`.

```
Erlang/OTP 18 [erts-7.2.1] [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.2.3
```

**Comment by josevalim** [comment]
The build is running on Elixir v1.0.2 where this won't work. I will have to investigate later then. Thank you!

**Comment by bjfish** [comment]
In case this information helps, I did a matrix build in my fork. Here are the results: https://travis-ci.org/bjfish/ex_doc/builds/118281832

|Version  | Result |
| ------------- | ------------- |
|  Elixir: 1.0.2 OTP Release: 17.4  | :x:  |
|  Elixir: 1.0.2 OTP Release: 18  | :x:  |
|  Elixir: 1.1.1 OTP Release: 17.4  | :white_check_mark:  |
|  Elixir: 1.1.1 OTP Release: 18 | :white_check_mark:  |
|  Elixir: 1.2.3 OTP Release: 18  | :white_check_mark:  |

**Comment by josevalim** [comment]
Excellent. We should run this test only on Elixir v1.1 because @callback does not matter for earlier Elixir version. :)

**Comment by bjfish** [comment]
Not sure if there was a better way to exclude/include tests by version than this update.

One nice thing in ruby spec is 
```
ruby_version_is "2.3" do
end
```

Alternatively the `if` statement could have gone in the fixture to have `@callback` or `defcallback` used by version.

Also, I could add the matrix build if you would like.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**